napalm_nxos/nxos.py

```diff
@@ -241,6 +241,46 @@ def get_lldp_neighbors(self):
 
         return results
 
+
+    def get_bgp_neighbors(self):
+        cmd = 'show bgp sessions vrf all'
+        vrf_list = self._get_command_table(cmd, 'TABLE_vrf', 'ROW_vrf')
```

Comment:
Again, this is going to fail in a single-VRF context. As a general rule: where you'd expect to be a list, when the list should contain one single element, crisco will give you a dict instead:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "napalm_nxos/nxos.py", line 252, in get_bgp_neighbors
    result_vrf_dict['router_id'] = unicode(vrf_dict['router-id'])
```

Please do again that trick:

```python
vrf_list = self._get_command_table(cmd, 'TABLE_vrf', 'ROW_vrf')
if isinstance(vrf_list, dict):
    vrf_list = [vrf_list]
```